src/BookmarkWindow.cpp

```diff
@@ -56,28 +54,38 @@ TBookmarkWindow::TBookmarkWindow(float x, float y, const char *title, const char
 	
 	
 	
-	fSaveAndConnectButton = new BButton(BRect(10, 215, 140, 240), "Connect", B_TRANSLATE("Connect"), new BMessage(CONNECT_CLICKED));
+	fSaveAndConnectButton = new BButton("Connect", B_TRANSLATE("Connect"), new BMessage(CONNECT_CLICKED));
 	fSaveAndConnectButton->SetEnabled(false);
 	
-	fSaveButton = new BButton(BRect(150, 215, 210, 240), "Save", B_TRANSLATE("Save"), new BMessage(SAVE_CLICKED));
+	fSaveButton = new BButton("Save", B_TRANSLATE("Save"), new BMessage(SAVE_CLICKED));
 	fSaveButton->SetEnabled(false);
 	
-	BButton *cancelButton = new BButton(BRect(220, 215, 280, 240), "Cancel", kCancel, new BMessage(B_QUIT_REQUESTED));
-	
-	BLayoutBuilder::Group<>(bgview,B_VERTICAL,2)
-		.Add(fBookmarkName)
-		.Add(fHostAddress)
-		.Add(fPort)
-		.Add(fUsername)
-		.Add(fPassword)
-		.Add(fEncoder)
-		.Add(fRemotePath)
-		.Add(fLocalPath)
-		.AddGroup(B_HORIZONTAL,10)
+	BButton *cancelButton = new BButton("Cancel", kCancel, new BMessage(B_QUIT_REQUESTED));
+	
+	BLayoutBuilder::Grid<>(this,2,2)
+		.SetInsets(10)
+		.Add(fBookmarkName->CreateLabelLayoutItem(),0,0)
+		.Add(fBookmarkName->CreateTextViewLayoutItem(),1,0)
+		.Add(fHostAddress->CreateLabelLayoutItem(),0,1)
+		.Add(fHostAddress->CreateTextViewLayoutItem(),1,1)
+		.Add(fPort->CreateLabelLayoutItem(),0,2)
+		.Add(fPort->CreateTextViewLayoutItem(),1,2)
+		.Add(fUsername->CreateLabelLayoutItem(),0,3)
+		.Add(fUsername->CreateTextViewLayoutItem(),1,3)
+		.Add(fPassword->CreateLabelLayoutItem(),0,4)
+		.Add(fPassword->CreateTextViewLayoutItem(),1,4)
+		.Add(fEncoder->CreateLabelLayoutItem(),0,5)
+		.Add(fEncoder->CreateTextViewLayoutItem(),1,5)
+		.Add(fRemotePath->CreateLabelLayoutItem(),0,6)
+		.Add(fRemotePath->CreateTextViewLayoutItem(),1,6)
+		.Add(fLocalPath->CreateLabelLayoutItem(),0,7)
+		.Add(fLocalPath->CreateTextViewLayoutItem(),1,7)
+		.AddGroup(B_HORIZONTAL,10,0,8,2)
```

Comment:
Could add a .AddStrut(B_USE_BIG_SPACING) in front of the buttons.

Reply:
Changes done

Reply:
The strut should be before the horizontal group.